scripts/hook_injector.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+'''
+hook_injector.py
+
+    This example harnesses the power of
+    Manticore as a debugger during program analysis,
+    providing introspection of hooks injected through
+    user-specified pattern inputs.
+
+'''
+import re
+import argparse
+
+from manticore import Manticore
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+
+
+class Introspector(object):
+    """
+    Introspector object is used to correctly parse
+    an ELF binary with a symbol table
+    """
+
+    def __init__(self, target_f, pattern):
+
+        # collected mem addresses from symbol pattern
+        self.functions = []
+
+        # instantiate manticore with binary target
+        self.mcore = Manticore(target_f)
+
+        with open(target_f, 'rb') as binary:
+
+            # parse binary
+            elf = ELFFile(binary)
+            if elf is None:
+                raise Exception("symbols are unsupported")
+
+            # retrieve symbols from all sections in binary
+            for section in elf.iter_sections():
```

Comment:
Loud thinking: maybe we shall move this (yielding/generating sections or symbols or returning a list of them?) into `Manticore` and use such method also in here https://github.com/trailofbits/manticore/blob/master/manticore/manticore.py#L655-L676 ?